Mohammad Idham

midham14 opened the pull request:
ready :
server:
- api documentation
 - get todo data
 - post todo data
- get findbyid
- put todo
- patch todo
- delete todo
-post login
-post register
- authentication
- authorization
- 3rd api & error handler blm 100%
- login google server
- validate password server ( sementara hanya length )

client :
-login dan get data via client menggunakan jquery
- register client
- login & logout google client
- add todo client
- edit status todo client
-  delete todo client




on progress :
- rapihin error handle yang belom dikirim ke middleware errorHandler()
- add styling client

midham14 committed:
api documentation & get todo data & add todo data

midham14 committed:
add findbyid,put,patch,delete

midham14 committed:
login & register | Authentication & Authorization

ziadyjo commented on models/todo.js:
```diff
@@ -0,0 +1,30 @@
+    title: DataTypes.STRING,
+    description: DataTypes.STRING,
+    status: DataTypes.STRING,
+    due_date: DataTypes.DATE
```
due datenya dihandle ya, tidak boleh diinputkan hari sebelum hari ini

ziadyjo commented on models/user.js:
```diff
@@ -0,0 +1,44 @@
+       }
+     }
+    },
+    password: DataTypes.STRING
```
password dihandle sesuai requirement ya

ziadyjo commented on routes/TodoRouter.js:
```diff
@@ -0,0 +1,14 @@
+
+router.get('/', Authentication  ,TodoController.getTodoHandler)
+router.post('/', Authentication  , TodoController.createTodoHandler)
+router.get('/:id', Authentication, Authorization  ,TodoController.getTodoByIdHandler)
```
seharusnya untuk get by id tidak perlu authorization ya

midham14 replied:
ok siap kak, nanti semua feedbacknya di perbaiki

midham14 committed:
errorhandler & 3rd api blm 100%, login dan get data client/jquery

midham14 committed:
1. add register client
2. add login & logout google client
3. add tambah todo
4. add edit status todo
5. add delete todo
6. validate password server ( sementara hanya length )
7. add login google server

midham14 committed:
1. add validasi due date
2. clear input field after add
3. add unique to email
